cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -79,14 +80,29 @@ func NewCertificateAuthority(config *CertConfig) (*x509.Certificate, crypto.Sign
 	return cert, key, nil
 }
 
+// NewIntermediateCertificateAuthority creates new certificate and private key for an intermediate certificate authority
+func NewIntermediateCertificateAuthority(parentCert *x509.Certificate, parentKey crypto.Signer, config *CertConfig) (*x509.Certificate, crypto.Signer, error) {
```

Comment:
config should be the first argument.

maybe the better API here is to have a single NewCertificateAuthority function that decides on self-signed vs signed-by-parent depending on a `nil` `parentCert`, but having a separate function is fine too.

Reply:
Change made in this commit: https://github.com/kubernetes/kubernetes/pull/97266/commits/6f90532f2db098c52a531bb17e6db39d23a8db41

Since creating an intermediate CA is an uncommon use case (it's only used in the tests for the certificate chain verification) I think it's preferable not to force callers of `NewCertificateAuthority` (of which there are quite a few) to have to specify a `nil` certificate and key.